js/src/client.ts

```diff
@@ -2148,6 +2162,80 @@ export class Client {
     await response.json();
   }
 
+  public async indexDataset({
+    datasetId,
+    datasetName,
+    tag,
+  }: {
+    datasetId?: string;
+    datasetName?: string;
+    tag?: string;
+  }): Promise<void> {
+    let datasetId_ = datasetId;
+    if (datasetId_ === undefined && datasetName === undefined) {
+      throw new Error("Must provide either datasetName or datasetId");
+    } else if (datasetId_ !== undefined && datasetName !== undefined) {
```

Comment:
I would do `if (!datasetId_ && ...)` instead of undefined checks because someone could pass in two empty strings, which would pass the `undefined` checks but not actually be defined

Same idea for the other checks too